homeassistant/components/sensor/gitlab_ci.py

```diff
@@ -0,0 +1,182 @@
+    _url = config.get(CONF_URL)
+
+    _gitlab_data = GitLabData(
+        priv_token=_priv_token,
+        gitlab_id=_gitlab_id,
+        interval=_interval,
+        url=_url
+    )
+
+    add_devices([GitLabSensor(_gitlab_id, _priv_token, _gitlab_data)], True)
+
+
+class GitLabSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, gitlab_id, priv_token, gitlab_data):
+        """Initialize the sensor."""
+        self._state = None
+        self._available = False
+        self._status = None
+        self._started_at = None
+        self._finished_at = None
+        self._duration = None
+        self._commit_id = None
+        self._commit_date = None
+        self._build_id = None
+        self._branch = None
+        self._gitlab_data = gitlab_data
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return 'GitLab CI Status'
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._available
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            ATTR_BUILD_STATUS: self._status,
+            ATTR_BUILD_STARTED: self._started_at,
+            ATTR_BUILD_FINISHED: self._finished_at,
+            ATTR_BUILD_DURATION: self._duration,
+            ATTR_BUILD_COMMIT_ID: self._commit_id,
+            ATTR_BUILD_COMMIT_DATE: self._commit_date,
+            ATTR_BUILD_ID: self._build_id,
+            ATTR_BUILD_BRANCH: self._branch
+        }
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        if self._status == 'success':
+            return ICON_HAPPY
+        if self._status == 'failed':
+            return ICON_SAD
+        return ICON_OTHER
+
+    def update(self):
+        """Collect updated data from GitLab API."""
+        self._gitlab_data.update()
+
+        self._status = self._gitlab_data.status
+        self._started_at = self._gitlab_data.started_at
+        self._finished_at = self._gitlab_data.finished_at
+        self._duration = self._gitlab_data.duration
+        self._commit_id = self._gitlab_data.commit_id
+        self._commit_date = self._gitlab_data.commit_date
+        self._build_id = self._gitlab_data.build_id
+        self._branch = self._gitlab_data.branch
+        self._state = self._gitlab_data.status
+        self._available = self._gitlab_data.available
+
+
+class GitLabData():
+    """GitLab Data object."""
+
+    def __init__(self, gitlab_id, priv_token, interval, url):
+        """Fetch data from GitLab API for most recent CI job."""
+        import gitlab
+        self._gitlab_id = gitlab_id
+        self._priv_token = priv_token
+        self._interval = interval
```

Comment:
Remove this. We only use it for a debug log, which doesn't seem necessary either.

Reply:
Removed.